### PR TITLE
Add constant to use core cron

### PR DESCRIPTION
**Pull request by davidsword**
## Description

Add constant to use core cron, instead of cron-control.

May be useful for huge multi sites that run only a few smaller jobs.

Bailing as early as possible. Using a constant instead of filters due to how early this file is.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [x] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

1. Put some xdebug break points in cron-control/cron-control.php plugin
1. load site, notice cron-control.php loads
1. add `define( 'WPCOM_VIP_USE_CORE_CRON', true );` to vip-config
1. reload site, notice cron-control.php doesn't load
1. check the cron queue, note upcoming or due events
1. wait a few minutes, refresh site a few times (reminder to bust the cache)
1. check cron queue again, that jobs moved from core cron working

### CODE REVIEW

**Comment by davidsword** [comment]
Setting as draft until I write a test for it, and check if any docs should be updated about this.

**Comment by davidsword** [comment]
No tests as `wpcom_vip_use_core_cron()` is `true` for local dev (phpunit tests) anyways.

imho we don't need any front-facing documentation because CronControl is the correct solution the vast majority of the time. Having reverting to core in docs may lead to disabling instead of properly troubleshooting.

File: 001-cron.php [hunk 1]
```diff
@@ -16,6 +16,11 @@
  * @return bool
  */
 function wpcom_vip_use_core_cron() {
+
+	if ( defined( 'WPCOM_VIP_USE_CORE_CRON' ) && WPCOM_VIP_USE_CORE_CRON ) {
```
Comment (mjangda) [inline 001-cron.php]: We've started to drop the `WPCOM_` prefix. Plus, add a disclaimer.

```suggestion
	// For some applications, Cron Control is not a good fit. Allow those apps to use core cron instead. 
	// Note: please check with the VIP team before using this constant as it's not always guaranteed to work.
	if ( defined( 'VIP_USE_CORE_CRON' ) && VIP_USE_CORE_CRON ) {
```

**Comment by mjangda** [comment]
@davidsword Actually, on testing this out, I realized that switching between Cron Control and core cron will result in any one-time scheduled events (like scheduled posts) to be lost, since those are stored in the Cron Control table and not options. That seems like a pretty big blocker for this. I don't have any good answers or workaround either right now but will think on it.

**Comment by joshbetz** [comment]
Most likely will be fixed by https://github.com/Automattic/Cron-Control/pull/188 when it's deployed.

**Pull request closed without merging**